pkg/apiserver/user/auth.go

```diff
@@ -54,34 +56,37 @@ type TokenResponse struct {
 	Expire time.Time `json:"expire"`
 }
 
-func (f *authenticateForm) Authenticate(tidbForwarder *tidb.Forwarder) (*utils.SessionUser, error) {
-	// TODO: Support non TiDB auth
-	if !f.IsTiDBAuth {
-		return nil, ErrSignInUnsupportedAuthType.New("unsupported auth type, only TiDB auth is supported")
-	}
-	db, err := tidbForwarder.OpenTiDB(f.Username, f.Password)
-	if err != nil {
-		if errorx.Cast(err) == nil {
+func (f *authenticateForm) Authenticate(tidbForwarder *tidb.Forwarder, etcdClient *clientv3.Client) (*utils.SessionUser, error) {
+	if !*f.IsTiDBAuth {
+		err := kvauth.VerifyKvAuthKey(etcdClient, f.Username, f.Password)
+		if err != nil {
 			return nil, ErrSignInOther.WrapWithNoMessage(err)
```

Comment:
`ErrSignInOther.WrapWithNoMessage(err)` is not an appropiate error code. If you are already responding an errorx in `VerifyKvAuthKey`, then you can follow L67 ~ L77. Actually they can be extracted to the outside to share the common logic.